### ISSUE TITLE
[java] UselessPureMethodCall false positive with AtomicInteger::getAndIncrement

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#6055]
**Affects PMD Version:** 7.17.0

**Rule:** [UselessPureMethodCall](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#uselesspuremethodcall)

**Description:**

AtomicInteger#getAndIncrement() has a side-effect, so it should be ok to ignore the result and the method is not useless.

**Code Sample demonstrating the issue:**

```java
callCount.getAndIncrement();
```
https://github.com/pmd/pmd/blob/7b142249a406d3f34256afb73f789055f7188671/pmd-core/src/main/java/net/sourceforge/pmd/benchmark/TimeTracker.java#L212

**Expected outcome:**

PMD reports a violation, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Comment by zbynek** [comment]
A workaround is to use `incrementAndGet` instead (since the result is ignored, the order doesn't matter).
I filed #6082  to fix this.
